pkg/policy/validate.go

```diff
@@ -133,6 +134,14 @@ func Validate(policy *kyverno.ClusterPolicy, client *dclient.Client, mock bool,
 			return fmt.Errorf("wildcards (*) are currently not supported in the match.resources.kinds field. at least one resource kind must be specified in a kind block.")
 		}
 
+		// Validate Kind with match resource kind's
+		for _, kind := range rule.MatchResources.Kinds {
+			_, k := c.GetKindFromGVK(kind)
+			if k == p.Kind {
+				return fmt.Errorf(" kind and match resource kind should not be same")
```

Comment:
(remove space between opening double quotes and "kind")
"kind and match resource kind should not be the same."